app/controllers/detainee_details_controller.rb

```diff
@@ -44,17 +34,41 @@ def update
 
   private
 
-  def redirect_to_move_or_profile
+  def redirect_if_detainee_exists
+    return unless params[:prison_number].present?
+    @_detainee = Detainee.find_by_prison_number(params[:prison_number])
+    return unless @_detainee
+    redirect_to_move_or_profile(flash: { warning: t('alerts.detainee_already_exists') })
+  end
+
+  def redirect_to_move_or_profile(options = {})
     if active_move.present?
-      redirect_to profile_path(active_move)
+      redirect_to profile_path(active_move), options
     elsif detainee.moves.any?
-      redirect_to copy_move_path(detainee)
+      redirect_to copy_move_path(detainee), options
     else
-      redirect_to new_move_path(detainee)
+      redirect_to new_move_path(detainee), options
     end
   end
 
   def form
     @_form ||= Forms::DetaineeDetails.new(detainee)
   end
+
+  def detainee_default_attrs
+    return {} unless params[:prison_number].present?
```

Comment:
We can switch the if not's to if's using `blank?` - think this applies to a few places.